src/WasmModule.ts

```diff
@@ -0,0 +1,206 @@
+import { decode } from '@webassemblyjs/wasm-parser';
+import ast from '@webassemblyjs/ast';
+
+import Graph from './Graph';
+import { IdMap, ModuleJSON } from './rollup/types';
+
+import { makeLegal } from './utils/identifierHelpers';
+import { basename, extname } from './utils/path';
+
+import ExternalModule from './ExternalModule';
+import Module from './Module';
+
+import ModuleScope from './ast/scopes/ModuleScope';
+import Variable from './ast/variables/Variable';
+import NamespaceVariable from './ast/variables/NamespaceVariable';
+
+const decoderOpts = {
+	ignoreCodeSection: true,
+	ignoreDataSection: true
+};
+
+type BuildLoadArgs = {
+	NAME: string;
+	URL: string;
+	IMPORT_OBJECT?: string;
+};
+
+const buildLoader = ({ NAME, URL, IMPORT_OBJECT }: BuildLoadArgs) => `
+	// function then$${NAME}(resolve) {
+	function then(resolve) {
+		if (typeof WebAssembly.instantiateStreaming !== 'function') {
+		  throw new Error('WebAssembly.instantiateStreaming is not supported');
+		}
+
+		if (typeof window.fetch !== 'function') {
+		  throw new Error('window.fetch is not supported');
+		}
+
+		const req = window.fetch('${URL}');
+
+		WebAssembly
+			.instantiateStreaming(req, ${IMPORT_OBJECT || '{}'})
+			.then(res => res.instance.exports)
+			.then(resolve)
+			.catch(resolve);
+	}
+`;
```

Comment:
Can this be treated as the "dynamic import mechanism" as is used in the JS Chunk.ts file that does the import to the associated wasmChunk.ts file?

Reply:
(and let me know if that would handle the issue you raised now?)

Reply:
I guess it would need to be a `wasmImport` or similar mechanism associated with the dynamic import specifically in the wasm case.

The point being that we are handling this in the `src/ast/nodes/Import.ts` render itself as the dynamic import rendering.

Reply:
Sorry but i'm really not sure to understand that.

> "dynamic import mechanism"

Yes, that's the code behind `import`.

> (and let me know if that would handle the issue you raised now?)

You mean the binding conflict issue?